src/OpenLoco/src/Vehicles/Vehicle1.cpp

```diff
@@ -244,14 +248,580 @@ namespace OpenLoco::Vehicles
+        auto routingRing = RoutingManager::RingView(veh1.routingHandle);
+        auto iter = ++routingRing.begin();
+        constexpr auto kResetRouting = static_cast<uint16_t>(~((1U << 7) | (1U << 8)));
+        for (auto i = 0; i < numRoadPieces; ++iter, ++i)
+        {
+            if (iter == routingRing.end())
+            {
+                break;
+            }
+            const auto routing = RoutingManager::getRouting(*iter);
+            if (i == 0)
+            {
+                RoutingManager::setRouting(*iter, (routing & kResetRouting) | (1U << 8));
+            }
+            else if (i == (numRoadPieces - 1))
+            {
+                RoutingManager::setRouting(*iter, (routing & kResetRouting) | (1U << 8) | (1U << 7));
+            }
+            else
+            {
+                RoutingManager::setRouting(*iter, (routing & kResetRouting) | (1 << 7));
+            }
+        }
+
+        const auto newDistance = vehicle1UpdateRoadMotionByPieces(veh1, numRoadPieces);
+        veh1.var_3C += newDistance;
+        auto* head = EntityManager::get<VehicleHead>(veh1.head);
+        head->var_3C += newDistance - noOvertakeDistance;
+    }
+
+    // 0x0047D52B
+    // veh1 : esi
+    // numRoadPieces : ah
+    static void applyChangeLaneToVehicle1(Vehicle1& veh1, uint8_t numRoadPieces)
+    {
+        const auto noOvertakeDistance = vehicle1UpdateRoadMotionByPiecesNoMove(veh1, numRoadPieces);
+        auto routingRing = RoutingManager::RingView(veh1.routingHandle);
+        auto iter = ++routingRing.begin();
+        constexpr auto kResetRouting = static_cast<uint16_t>(~((1U << 7) | (1U << 8)));
+        for (auto i = 0; i < numRoadPieces; ++iter, ++i)
+        {
+            if (iter == routingRing.end())
+            {
+                break;
+            }
+            const auto routing = RoutingManager::getRouting(*iter);
+            if (i == 0)
+            {
+                RoutingManager::setRouting(*iter, (routing & kResetRouting) | (1U << 8));
+            }
+            else
+            {
+                RoutingManager::setRouting(*iter, (routing & kResetRouting) | (1 << 7));
+            }
+        }
+
+        const auto newDistance = vehicle1UpdateRoadMotionByPieces(veh1, numRoadPieces);
+        veh1.var_3C += newDistance;
+        auto* head = EntityManager::get<VehicleHead>(veh1.head);
+        head->var_3C += newDistance - noOvertakeDistance;
+        head->trackAndDirection.road._data &= kResetRouting;
+        head->trackAndDirection.road._data |= numRoadPieces <= 1 ? (1U << 8) : (1U << 7);
+    }
+    struct Sub47CEB7Result
+    {
+        uint8_t al;
+        uint8_t ah;
+    };
+
+    // 0x0047CEB7
+    // veh1 : esi
+    // return : al, ah
+    static Sub47CEB7Result sub_47CEB7(Vehicle1& veh1)
+    {
+        const auto startPos = World::Pos3(veh1.tileX, veh1.tileY, veh1.tileBaseZ * World::kSmallZStep);
+        uint8_t unk112C327 = 0;
```

Comment:
distance/iterations, not entirely sure given its called numRoadPieces elsewhere.